crates/forge-runner/src/build_trace_data/test_program_path.rs

```diff
@@ -0,0 +1,46 @@
+use anyhow::Context;
+use anyhow::Result;
+use cairo_lang_sierra::program::VersionedProgram;
+use camino::Utf8PathBuf;
+use fs4::FileExt;
+use std::fs;
+use std::fs::File;
+use std::io::BufWriter;
+
+pub const TESTS_PROGRAMS_DIR: &str = ".snfoundry_test_code";
+
+#[derive(Clone)]
+pub struct TestProgramPath(Utf8PathBuf);
+
+impl TestProgramPath {
+    pub fn save_test_program(
```

Comment:
This shouldn't be a function of `TestProgramPath` imo, should be a free function. It suggests that it saves program under `TestProgramPath`.

Reply:
Can change it, wanted it to be in one function to guarantee that the path is valid

Reply:
Refactored it a bit so the constructor also generates the path, guaranteeing its uniqueness

Reply:
Changed the logi chere